github-auth-provider/pkg/profile/profile.go

```diff
@@ -43,27 +49,83 @@ type githubResponse struct {
 	UpdatedAt         string `json:"updated_at"`
 }
 
-func FetchGitHubProfile(ctx context.Context, accessToken, url string) (*githubResponse, error) {
-	req, err := http.NewRequestWithContext(ctx, http.MethodGet, url, nil)
+type githubOrganization struct {
+	ID        int64  `json:"id"`
+	Login     string `json:"login"`
+	AvatarURL string `json:"avatar_url"`
+}
+
+type githubTeam struct {
+	ID           int64              `json:"id"`
+	Name         string             `json:"name"`
+	Organization githubOrganization `json:"organization"`
+}
+
+func FetchUserProfile(ctx context.Context, accessToken string) (*githubUserProfile, error) {
+	result, err := makeGitHubRequest[githubUserProfile](ctx, accessToken, "user")
```

Comment:
pico-nit: not a big var of this type of generics usage because you have to explicitly state the type parameter. In my opinion, it is less readable than just doing:
```suggestion
	var result githubUserProfile
	err := makeGitHubRequest[githubUserProfile](ctx, accessToken, "user", &result)
```

Then the signature of `makeGitHubRequest` would change to `func makeGitHubRequest(ctx context.Context, accessToken, path string, result any) error` and none of the implementation of the function changes.

Reply:
fair enough. I'll update the signature and usage

Reply:
fixed